### PR TITLE
[DSC_PARSER] Keep track of all the paths that we find in a DSC

### PR DESCRIPTION
**Pull request by matthewfcarlson**
When parsing a DSC, we should keep track of all the files that are included in a DSC.

### CODE REVIEW

File: edk2toollib/uefi/edk2/parsers/dsc_parser_test.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# @file dsc_parser_test.py
```
Comment (corthon) [inline edk2toollib/uefi/edk2/parsers/dsc_parser_test.py]: This file seems unrelated...

**Comment by spbrogan** [comment]
In the commit message can you write more details.  We need to start doing better on our commits especially when PR content is not obvious on what it does.

File: edk2toollib/uefi/edk2/parsers/dsc_parser_test.py (hunk 1 above)
Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/dsc_parser_test.py]: I'm modifying the DSC parser so I think it makes sense to add a test for it :) 
As it stands right now, testing !includes in the parser would be pretty hard without creating a bunch of files. I guess I could do it

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/dsc_parser_test.py]: Added a much better test

**Comment by matthewfcarlson** [comment]
Commits are trimmed down and fixed up
